fix(RHINENG-7963): Bug with select all CVEs option on CVEs page

LiorKGOW opened the pull request:
This PR addresses issue [#RHINENG-7963](https://issues.redhat.com/browse/RHINENG-7963)

When attempting to "select all" CVEs in the CVEs Page, none are selected
With this code selecting all CVEs works

The reason why this bug occurres:
If a user has Edge systems in their account, we need to convert the affecting param to be in affectingHostType 
So, we use hybrid system filter to determine this.
- If the user have both Conventional & Edge systems, we convert the affecting params to be in affectingHostType params because affecting is no longer being used in the backend in this case.
- If the user has only Conventional systems, the backend still uses affecting param, so there is no need for the conversion.

Because affecting param is no longer being used in the backend in the first case, sending it in the request would make the request fail, or in case the request does not fail, it would return bad results

Location where this 'selectAllCVEs' option is located:
![Location where this 'selectAllCVEs' option is located](https://github.com/RedHatInsights/vulnerability-ui/assets/93318917/a050ee60-d81c-4dbf-9f13-1ebce0cd90c4)

bastilian commented on src/Helpers/APIHelper.js:
```diff
@@ -118,7 +118,21 @@ export function getCveIdsList(apiProps) {
     ];
 
-    let parameterArray = constructParameters(apiProps, parameterNames);
+    /* If a user has Edge systems in their account, we need to convert the affecting param to be in affectingHostType
+    * So, we use hybrid system filter to determine this.
+    * - If the user have both Conventional & Edge systems, we convert the affecting params to be in affectingHostType params
+    * because affecting is no longer being used in the backend in this case.
+    * - If the user has only Conventional systems, the backend still uses affecting param, so there is no need for the conversion.
+    *
+    * Because affecting param is no longer being used in the backend in the first case,
+    * sending it in the request would make the request fail, or in case the request does not fail, it would return bad results
+    */
```
This is more something that could be put into the commit message or the PR.

The code itself should be expressive enough to not have to explain it.

LiorKGOW replied:
I will add it to the commit message 👍🏼

LiorKGOW commented:
Thank you for the review @bastilian ! 
I have addressed your comment and moved the comment in the code to be in the commit message instead 👍🏼 

This PR is ready for another review 😄

mkholjuraev commented on src/Helpers/APIHelper.js:
```diff
@@ -109,7 +109,7 @@ export function getCveListByAccount(apiProps, shouldUseHybridSystemFilter = fals
 }
 
-export function getCveIdsList(apiProps) {
+export function getCveIdsList(apiProps, shouldUseHybridSystemFilter = true) {
```
Should we remove this hardcoded 'true' and pass `shouldUseHybridSystemFilter` value from the function that is calling `getCveIdsList`?

LiorKGOW replied:
Of course :)

mkholjuraev commented on src/Helpers/APIHelper.js:
```diff
@@ -118,7 +118,12 @@ export function getCveIdsList(apiProps) {
-    let parameterArray = constructParameters(apiProps, parameterNames);
+    const apiPropsWithFlag = {
+        ...apiProps,
+        ...shouldUseHybridSystemFilter ? { affecting: undefined, affectingHostType: apiProps.affecting } : {}
```
Maybe we can place this bit of code to change the affecting filter key conditionally inside `constructParameters`. With this, we would make sure the correct filter key is used in all API fetch functions.

LiorKGOW replied:
Do you think it is part of the 'select all CVEs doesn't work' bug ? 
Maybe we can open a ticket for it and add these changes in another PR
What do you think @mkholjuraev ?

mkholjuraev replied:
I believe if we open a ticket, it will not get much attention due to other high-prior tasks. The changes should be very small, all we would need to do is move the code highlighted into `constructParameters `. If you think this does not give enough value, feel free to move on :)

LiorKGOW replied:
I have added your suggestion and added a 'false' default value for this argument (shouldUseHybridSystemFilter) inside constructParameters
So it won't effect other actions, only the one we are changing in this PR (getCveIdsList) 👍🏼

LiorKGOW commented:
/retest

mkholjuraev reviewed (approved):
LGTM! Tested locally, I was able to select all items using an account with both conventional and edge systems.

LiorKGOW commented:
Thank you for the review ! @mkholjuraev 
I have addressed your comments so the PR is ready for another review :)

mkholjuraev commented on src/Helpers/APIHelper.js:
```diff
@@ -109,16 +109,17 @@ export function getCveListByAccount(apiProps, shouldUseHybridSystemFilter = fals
 }
 
-export function getCveIdsList(apiProps) {
+export function getCveIdsList(apiProps, shouldUseHybridSystemFilter) {
```
It seems your changes are missing. Can we do the same with `getCveListByAccount`?

LiorKGOW replied:
Sure
I have added those changes 👍🏼

LiorKGOW commented:
Rebased and repushed after #2079 was merged to master

mkholjuraev commented:
@LiorKGOW looks good to me now. Ready to be merged now
